cypress/integration/pages/onDemandRadio/tests.js

```diff
@@ -53,5 +54,74 @@ export default ({ service, pageType, variant, isAmp }) => {
         });
       });
     });
+    describe(`Tests for ${service} ${pageType} ${variant} with toggle use`, () => {
+      before(() => {
+        cy.getToggles(service);
+      });
+      describe('Recent Episodes component', () => {
+        it('should be displayed if the toggle is on, and shows the expected number of items (does not run on local)', function test() {
+          // I can enable tests running on local when I have updated all the fixture data to show the component
+          if (Cypress.env('APP_ENV') === 'local') {
+            cy.log('Does not run on local');
+          } else {
+            cy.fixture(`toggles/${service}.json`).then(toggles => {
+              const recentEpisodesEnabled = path(
+                ['recentAudioEpisodes', 'enabled'],
+                toggles,
+              );
+              cy.log(
+                `Recent Episodes component enabled? ${recentEpisodesEnabled}`,
+              );
+              // There cannot be more episodes shown than the max allowed
+              if (recentEpisodesEnabled) {
+                const recentEpisodesMaxNumber = path(
+                  ['recentAudioEpisodes', 'value'],
+                  toggles,
+                );
+                const currentPath = Cypress.env('currentPath');
+                const url =
+                  Cypress.env('APP_ENV') === 'test'
+                    ? `${currentPath}?renderer_env=live`
+                    : `${currentPath}`;
+
+                cy.request(getDataUrl(url)).then(({ body }) => {
+                  // Count the number of episodes that are available and so will show (there can be unavailable episodes in the list)
+                  const expectedNumberOfEpisodes = body.relatedContent.groups[0].promos
+                    .filter(({ media }) => media.versions.length)
+                    .slice(0, recentEpisodesMaxNumber).length;
+
+                  cy.log(
+                    `Number of available episodes? ${expectedNumberOfEpisodes}`,
+                  );
+                  if (expectedNumberOfEpisodes > 1) {
+                    cy.get('[data-e2e=recent-episodes-list]').should('exist');
+
+                    cy.get('[data-e2e=recent-episodes-list]').within(() => {
+                      cy.get('[data-e2e=recent-episodes-list-item]')
+                        .its('length')
+                        .should('eq', expectedNumberOfEpisodes);
+                    });
+                  }
+                  // If there is only one item, it is not in a list
+                  else if (expectedNumberOfEpisodes === 1) {
+                    cy.get("div[class*='css-1sel12u-Wrapper emzt7w80']").should(
```

Comment:
Some of these selectors are too specific - bits of this could well change: `1sel12u` & `emzt7w80` in particular.

There's some reasonable explanation of how different selectors work here https://www.w3schools.com/cssref/css_selectors.asp (scroll down to [attribute] to find the ones we're using a lot in cypress)

`class*=` which you're using here means 'select every element where the class contains this substring'. This is a super broad selector, but by passing in the entire className, it's actually super specific.

Reply:
This will be changed when we have the data-e2e tags put into the tv component. They are not there yet.

Reply:
So regardless of whether we are able to use a data-e2e value for these - using the emotion labels (`css-1sel12u` & `emzt7w80`) in the selector turns this into a bit of a time bomb (_drama_). 

When the component has its styles changed (either in Simorgh, or Psammead) - these will be regenerated, and the class selector will no longer find the element.

There are examples of using selectors without data-e2e attributes elsewhere in Simorgh, eg: https://github.com/bbc/simorgh/blob/e470060cd6a1853cd58cb18435f02c653de097bc/src%2Fapp%2Fcontainers%2FCpsFeaturesAnalysis%2Findex.test.jsx#L92

So even if we don't have data-e2e values we can use, we can still use other selectors, without including the emotion-assigned labels.